src/components/HTMLEngineProvider/htmlEngineUtils.js

```diff
@@ -28,20 +28,41 @@ function isCommentTag(tagName) {
 }
 
 /**
- * Check if there is an ancestor node with name 'comment'.
- * Finding node with name 'comment' flags that we are rendering a comment.
+ * Check if there is an ancestor node for which the predicate returns true.
+ *
  * @param {TNode} tnode
+ * @param {Function} predicate
  * @returns {Boolean}
  */
-function isInsideComment(tnode) {
-    let currentNode = tnode;
-    while (currentNode.parent) {
-        if (isCommentTag(currentNode.domNode.name)) {
+function isChildOfNode(tnode, predicate) {
+    let currentNode = tnode.parent;
+    while (currentNode) {
+        if (predicate(currentNode)) {
             return true;
         }
         currentNode = currentNode.parent;
     }
     return false;
 }
 
-export {computeEmbeddedMaxWidth, isInsideComment, isCommentTag};
+/**
+ * Check if there is an ancestor node with name 'comment'.
+ * Finding node with name 'comment' flags that we are rendering a comment.
+ * @param {TNode} tnode
+ * @returns {Boolean}
+ */
+function isChildOfComment(tnode) {
+    return isChildOfNode(tnode, (node) => isCommentTag(node.domNode.name));
+}
+
+/**
+ * Check if there is an ancestor node with the name 'h1'.
+ * Finding a node with the name 'h1' flags that we are rendering inside an h1 element.
+ * @param {TNode} tnode
+ * @returns {Boolean}
+ */
+function isChildOfH1(tnode) {
+    return isChildOfNode(tnode, (node) => node.domNode.name.toLowerCase() === 'h1');
```

Comment:
FYI, this is causing staging to crash and I cannot load the page at all.

![image](https://github.com/Expensify/App/assets/1228807/2d829545-bcc6-4e1d-ba06-794f81e60744)

Reply:
@yakupafsin Can you please raise a quick PR fixing this? I think we may not need the `toLowerCase` afterall. The node name for html elements is expected to always be uppercase [^1].

[^1]: https://developer.mozilla.org/en-US/docs/Web/API/Node/nodeName

Reply:
Sorry for the disruption – this has been fixed: https://github.com/Expensify/App/pull/31079